### ISSUE TITLE
inconsistent interface assignment in BIG-IP

### ISSUE THREAD

**Issue by boreal321** [F5Networks/f5-google-gdm-templates#39]
## Do you already have an issue opened with F5 support?

Not yet but one is being opened by my customer.

## Description
Describe the problem you're having or the enhancement you'd like to request.

I deployed 1 device at a time using f5-existing-stack-byol-3nic-bigip GDM template and the issue is only eth0 is functional. The other 2 NICs disappear from the boxes for some strange reason after reboot. See my output below. Now you see eth1 & 2 and then you don't.

Steps taken:
Deploy VM
Ran ifconfig -a saw only eth0
I rebooted, ran the ifconfig and saw 3 NICs
A few seconds later I ran ifconfig and they were gone.
GCP VM details still lists 3 NICs

```
Console output:
Serial port 1 (console) output for bigip2-lb-deploy-01172020-lbl-1

SeaBIOS (version 1.8.2-20191106_143117-google)
Total RAM Size = 0x00000003c0000000 = 15360 MiB
CPUs found: 4     Max CPUs supported: 4
Comparing RSDP and RSDP
Comparing RSDT and RSDT
Comparing FACP and FACP
Comparing FACS and FACS
return 0 for FACS vs FACS: SUCCESS
Comparing DSDT and DSDT
return 0 for DSDT vs DSDT: SUCCESS
return 0 for FACP vs FACP: SUCCESS
Comparing SRAT and SRAT
return 0 for SRAT vs SRAT: SUCCESS
Comparing APIC and APIC
return 0 for APIC vs APIC: SUCCESS
Comparing SSDT and SSDT
return 0 for SSDT vs SSDT: SUCCESS
Comparing WAET and WAET
return 0 for WAET vs WAET: SUCCESS
return 0 for RSDT vs RSDT: SUCCESS
return 0 for RSDP vs RSDP: SUCCESS
found virtio-scsi at 0:3
virtio-scsi vendor='Google' product='PersistentDisk' rev='1' type=0 removable=0
virtio-scsi blksize=512 sectors=18874368 = 9216 MiB
drive 0x000f23d0: PCHS=0/0/0 translation=lba LCHS=1024/255/63 s=18874368
Sending Seabios boot VM event.
Booting from Hard Disk 0...
Red Hat nash version 5.1.19.6 starting
mdadm: No arrays found in config file or automatically
  Reading all physical volumes.  This may take a while...
  Found volume group "vg-db-sda" using metadata type lvm2
  9 logical volume(s) in volume group "vg-db-sda" now active
mdadm: No arrays found in config file or automatically
....
Starting plymouth daemon
INFO: Allocating 6738 2048kB hugepages
 
Starting bootchart logging
 
bootchart: no initrd used; starting
 
                                         Welcome to BIG-IP 13.1.3.2 Build 0.0.4
 
Starting udev: [  OK  ]
 
 
Set system time from hardware clock [  OK  ]
 
 
Setting hostname localhost.localdomain:  [  OK  ]
 
 
Run MD devices:  [  OK  ]
 
 
Setting up Logical Volume Management:   9 logical volume(s) in volume group "vg-db-sda" now active
 
[  OK  ]
 
 
Checking filesystems
 
Checking all file systems.
 
[/sbin/fsck.ext3 (1) -- /] fsck.ext3 -a /dev/mapper/vg--db--sda-set.1.root
 
set.1./: clean, 5834/112640 files, 292094/450560 blocks
 
[/sbin/fsck.ext3 (1) -- /shared] fsck.ext3 -a /dev/mapper/vg--db--sda-dat.share
 
dat.share: clean, 46/128016 files, 26738/512000 blocks
 
[/sbin/fsck.ext3 (1) -- /var/log] fsck.ext3 -a /dev/mapper/vg--db--sda-dat.log
 
dat.log: clean, 40/128016 files, 33814/512000 blocks
 
[/sbin/fsck.ext3 (1) -- /appdata] fsck.ext3 -a /dev/mapper/vg--db--sda-dat.appdata
 
dat.appdata: clean, 16/8192 files, 5555/32768 blocks
 
[/sbin/fsck.ext3 (1) -- /config] fsck.ext3 -a /dev/mapper/vg--db--sda-set.1._config
 
set.1./config: clean, 243/125984 files, 37679/503808 blocks
 
[/sbin/fsck.ext3 (1) -- /usr] fsck.ext3 -a /dev/mapper/vg--db--sda-set.1._usr
 
set.1./usr: clean, 82159/262944 files, 829785/1050624 blocks
 
[/sbin/fsck.ext3 (1) -- /var] fsck.ext3 -a /dev/mapper/vg--db--sda-set.1._var
 
set.1./var: clean, 8363/60928 files, 183855/243712 blocks
 
[  OK  ]
 
 
Remounting root filesystem in read-write mode:  [  OK  ]
 
 
Mounting local filesystems:  [  OK  ]
 
 
pid 1's current affinity mask: f
 
pid 1's new affinity mask: f
 
Enabling /etc/fstab swaps:  [  OK  ]
 
 
Auto-resize disks and directories:  [  OK  ]
 
Activating swap partition: [  OK  ]
 
 
Starting early syslog-ng: [  OK  ]
 
 
Virtual Edition customizations: [  OK  ]
 
net.ipv6.route.max_size = 536870911
 
mdadm: No arrays found in config file or automatically
 
info: mdadm returned code 256
 
Checking for and completing any logical disk transactions: [  OK  ]
 
 
Checking for and completing any disk management transactions: [  OK  ]
 
 
Setting up initial module provisioning: 'No modules have memory! Changing to initial boot configuration.'
 
[  OK  ]
 
 
Writing legacy mprov database: [  OK  ]
 
 
Starting increase_entropy: [  OK  ]
 
 
Starting jitterentropy-rngd: [  OK  ]
 
 
Entropy generation started successfully.
 
Exiting 22akeygen.sysinit
 
info: setting ownership of sda as mixed
 
info: setting ownership of sda as mixed
 
info: Looking for unbootable initial ramdisk images...
 
info: Looking for solid-state drives...
 
info: No solid-state drives needing hpa adjustment found in this system.
 
Entering non-interactive startup
 
Calling the system activity data collector (sadc)...
 
Starting monitoring for VG vg-db-sda:   9 logical volume(s) in volume group "vg-db-sda" monitored
 
[  OK  ]
 
 
ip6tables: Applying firewall rules: [  OK  ]
 
 
iptables: Applying firewall rules: [  OK  ]
 
 
Swapping ethernet interfaces:  mv: `/etc/sysconfig/network-scripts/ifcfg-eth0' and `/etc/sysconfig/network-scripts/ifcfg-eth0' are the same file
 
[  OK  ]
 
 
Bringing up loopback interface:  [  OK  ]
 
 
Bringing up interface dummy:  [  OK  ]
 
 
Bringing up interface eth0:  [  OK  ]
 
 
Bringing up interface mgmt: 
 
Determining IP information for mgmt... done.
 
[  OK  ]
 
 
Starting auditd: [  OK  ]
 
 
Starting mdmonitor: [  OK  ]
 
 
Starting syslog-ng: [  OK  ]
 
 
Starting pkg-tools [  OK  ]
 
 
Starting system message bus: [  OK  ]
 
 
Mounting filesystems:  [  OK  ]
 
 
Starting acpi daemon: [  OK  ]
 
 
Hyper-V drivers are not loaded.
 
Hyper-V KVP daemon not started:[WARNING]
 
 
Retrigger udev events[  OK  ]
 
 
Cloud detection successful.
 
Run all the deployment-specific startup scripts first
 
Cloud detection successful.
 
Starting Retrieve public keys for ssh(gce-init): [  OK  ]
 
Initializing cloud-init
 
vadc-init - DONE
 
Starting sshd:[  OK  ]
 
 
Starting ntpd: [  OK  ]
 
 
Current management-ip configuration mode is: DHCP.
 
Starting httpd: [  OK  ]
 
 
Starting crond: [  OK  ]
 ```

## Template

f5-existing-stack-same-net-cluster-byol-3nic-bigip.yaml

## Severity Level

2

**Comment by shyawnkarim** [comment]
Thanks for reaching out to us with this issue. I've gone ahead and created Jira ticket 1802 to fix this bug.

**Comment by shyawnkarim** [comment]
We believe this issue was fixed in [release 3.3.0](https://github.com/F5Networks/f5-google-gdm-templates/releases/tag/v3.3.0), which addressed configs getting lost after reboot.

 - Fixed an issue with standalone templates using 2 or 3 NICs that prevented NIC swap from completing until after a system reboot when using tmos version v13. 

 - Fixed an issue with failover templates that caused problems connecting to the management address after a reboot
